lib/pure/collections/sequtils.nim

```diff
@@ -918,6 +1024,500 @@ template newSeqWith*(len: int, init: untyped): untyped =
+    ##
+    ## The template injects the ``it`` variable which you can use directly in an
+    ## expression.
+    ##
+    block:
+      evalOnceAs(s1, s, compiles((let _ = s)))
+      s1[low(s1) ..< low(s1) + lenWhileIt(s1, op)]
+
+  template dropItWhile*(s: typed, op: untyped): untyped =
+    ## Returns a sequence of items in the container `s` without items that
+    ## keep fulfilling `op` (an expression resulting in a `bool`) from the beginning,
+    ## i.e. those items are dropped.
+    ##
+    ## The template injects the ``it`` variable which you can use directly in an
+    ## expression.
+    ##
+    block:
+      evalOnceAs(s1, s, compiles((let _ = s)))
+      s1[low(s1) + lenWhileIt(s1, op) ..< s1.len]
+
+  template nestIt*(op: untyped, x: typed, n: Natural): untyped =
+    ## Returns a new sequence of the results of evaluating `op` on previous value for `n`
+    ## times. The 0th item of the result is `x`.
+    ## `n` must be a non-negative number (zero or more).
+    ##
+    ## The template injects the ``it`` variable representing the previous value which
+    ## you can use directly in an expression.
+    ##
+    block:
+      var result = newSeq[typeof(x)](n + 1)
+      result[0] = x
+      for i in 1 .. n:
+        let it {.inject.} = result[i - 1]
+        result[i] = op
+      result
+
+  template findItFirst*(s: typed, op: untyped, def: typed): untyped =
+    ## Find the first element in `s` on which `op` evaluated to `true`. If such
+    ## element is not found is `s`, `def` is returned.
+    ##
+    ## The template injects the ``it`` variable representing the previous value which
+    ## you can use directly in an expression.
+    ##
+    block:
+      var result: typeof(def) = def
+      for it {.inject.} in s:
+        if op:
+          result = it
+          break
+      result
+
+  template findItFirst*(s: typed, op: untyped): untyped =
+    ## Find the first element in `s` on which `op` evaluated to `true`. If such
+    ## element is not found is `s`, `default(T)` is returned, where `T` is the
+    ## type of items in `s`.
+    ##
+    ## The template injects the ``it`` variable representing the previous value which
+    ## you can use directly in an expression.
+    ##
+    findItFirst(s, op, default(typeof(items(s), typeOfIter)))
+
+  template indexOfIt*(s: typed, op: untyped): int =
+    ## Get the index of the first element in `s` that fulfilled the predicate
+    ## `op` (a `bool` expression). If such element is not found is `s`,
+    ##  `low(s) - 1` is returned.
+    ##
+    ## The template injects the ``it`` variable representing the previous value which
+    ## you can use directly in an expression.
+    ##
+    block:
+      evalOnceAs(s1, s, compiles((let _ = s)))
+      var result = low(s1) - 1
+      for i in low(s1) .. high(s1):
+        let it {.inject.} = s1[i]
+        if op:
+          result = i
+          break
+      result
+
+  template scanIt*(s: typed, op: untyped) =
```

Comment:
IMHO this is a bit too silly to be added.

Reply:
IMHO, "do something on each item in a container" (i.e. `forEach` in quite some languages) is not
silly.

Reply:
Yes and we have that already and it's written as ``for``.

Reply:
Yes we have the `for` statement. Should we have two little function-like things (`scan`/`scanIt`)? 
It has something to do with "taste". It's all up to @Araq to keep the taste for Nim, saving
it from falling into chasm of silliness.

Getting tired, it seems better to withdraw this.

Reply:
Please address the raised points or withdraw the PR. IMHO we need to apply "If in doubt, leave it out" more often.